### DISCUSSION TITLE
Request to Add the ability to specify attributes, such as rel=nofollow

### DISCUSSION THREAD

**Discussion by kpm** [yuin/goldmark#370] [Ideas]
Hi,

Thanks for goldmark, I use it from hugo for my blog.  I'd like to suggest a feature, to allow custom attributes to be added to html elements such as links.   My use-case is to add rel="nofollow" to links, but I can imagine other elements might benefit too, for example adding a one-off class to an element.  I spent a lot of time trying to work out how to get this working, I think it would save others much time too if they ever need to customize attributes on their markdown.

There is an npm package which has similar features here:  https://www.npmjs.com/package/markdown-it-attrs 

Example input:
```
# header {.style-me}
paragraph {data-toggle=modal}
[this](/link){target="_blank"}
```

Example output:
```
<h1 class="style-me">header</h1>
<p data-toggle="modal">paragraph</p>
<a href="/link" target="_blank">this</a>
```

Thanks very much for considering this.

**Comment by jmooring** [comment]
@kpm

[Hugo](https://gohugo.io/) supports markdown attributes on images and block-level elements including blockquotes, fenced code blocks, headings, lists, paragraphs, and tables. Please see:
<https://discourse.gohugo.io/t/markdown-attributes/41783>

Attributes may be added to links using a markdown render hook.

If you have further questions, please create a topic on the [hugo forum](https://discourse.gohugo.io/).

**Comment by kpm** [comment]
Hi,

Thanks for that info.  However it doesn't support attributes on **links**, which is my use case.  I would say it's a pretty common use case too.  Shouldn't these things be supported at the level of the markdown processor?

**Comment by jmooring** [comment]
> Attributes may be added to links using a markdown render hook. 

And there's are couple of ways to do it.

If you have further questions, please create a topic on the [hugo forum](https://discourse.gohugo.io/).

**Comment by DenisDuev** [comment]
As there is already support in the goldmark for [headings](https://github.com/yuin/goldmark#attributes) can this attributes support be also extended for other elements?
